crates/oxc_linter/src/rules/typescript/no_useless_empty_export.rs

```diff
@@ -0,0 +1,175 @@
+use oxc_allocator::Vec;
+use oxc_ast::{
+    ast::{ExportNamedDeclaration, Statement, TSModuleDeclarationBody},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_useless_empty_export_diagnostic(span0: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(
+        "typescript-eslint(no-useless-empty-export): Disallow empty exports that don't change anything in a module file",
+    )
+    .with_help("Empty export does nothing and can be removed.")
+    .with_labels([span0.into()])
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessEmptyExport;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow empty exports that don't change anything in a module file.
+    ///
+    /// ### Example
+    ///
+    /// ### Bad
+    /// ```javascript
+    /// export const value = 'Hello, world!';
+    /// export {};
+    /// ```
+    ///
+    /// ### Good
+    /// ```javascript
+    /// export const value = 'Hello, world!';
+    /// ```
+    ///
+    NoUselessEmptyExport,
+    correctness
+);
+
+impl Rule for NoUselessEmptyExport {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::Program(program) => {
+                check_node(&program.body, ctx);
```

Comment:
It seems like we can add a fast path here.

Can you take a look at `ctx.semantic().module_record()` and see if there are any useful information that you can leverage?

Reply:
I have modified it to skip processing when it's `not_esm` or `export_default_duplicated`. Is this approach correct?
https://github.com/oxc-project/oxc/pull/3605/commits/790949fe2713d5d6bb7a25b0c90ad552c9879199